tests/devices/unit_tests/i24/test_pmac.py

```diff
@@ -88,6 +88,30 @@ async def go_high_then_low():
     assert await fake_pmac.pmac_string.get_value() == "&2b11r"
 
 
+@pytest.mark.parametrize("sleepTime, expected", [(0.05, 3), (0.2, 2)])
+async def test_counter_refresh_timeout(sleepTime, expected, fake_pmac: PMAC, RE):
+    async def update_counter():
+        set_mock_value(fake_pmac.scanstatus, 1)
+        set_mock_value(fake_pmac.counter, 0)
+        await asyncio.sleep(0.05)
+        set_mock_value(fake_pmac.counter, 1)
+        await asyncio.sleep(0.05)
+        set_mock_value(fake_pmac.counter, 2)
+        await asyncio.sleep(sleepTime)
+        set_mock_value(fake_pmac.counter, 3)
+        set_mock_value(fake_pmac.scanstatus, 0)
+
+    callback_on_mock_put(
+        fake_pmac.pmac_string,
+        lambda *args, **kwargs: asyncio.create_task(update_counter()),  # type: ignore
+    )
+
+    await fake_pmac.run_program.kickoff()
+    await fake_pmac.run_program.complete(counter_timeout=0.1)
```

Comment:
Must: I would expect this to test that if we hit the timeout it raised the expected exception